### PR TITLE
Closes #95 Closes #103 : removed test data and updated vignette to read data from pharmaversesdtm

### LINKED ISSUE #95: General Issue: Move test-data to pharmaversesdtm
### Background Information

Move test-data to pharmaversesdtm

### Definition of Done

_No response_

### LINKED ISSUE #103: Bug: derive_param_waisthip is throwing an error due to get_conv_factors_all
### What happened?

The assert_unit expects required_unit = "cm" but 5 values are being passed “cm” “m” “mm” “in” “ft” that's giving an error.

### Session Information

_No response_

### Reproducible Example
```
advs_check <- advs %>%
  derive_param_waisthip(
    by_vars = exprs(!!!get_admiral_option("subject_keys"), VISIT, VISITNUM),
    wstcir_code = "WSTCIR",
    hipcir_code = "HIPCIR",
    set_values_to = exprs(
      PARAMCD = "WAISTHIP",
      PARAM = "Waist to Hip Ratio"
    ),
    get_unit_expr = VSSTRESU
  )
```
produces

Error in `assert_unit()` at admiralmetabolic/R/derive_advs_params.R:143:2:
! Argument `required_unit` must be a scalar of class <character>, but is length 5
Backtrace:
 1. advs %>% ...
 2. admiralmetabolic::derive_param_waisthip(...)
 3. admiraldev::assert_unit(...)

Expected is 

units_supported <- names(get_conv_factors_all()[["length"]])

  assert_unit(
    dataset,
    param = wstcir_code,
    required_unit = "cm", [ but units_supported returns 5 values]
    get_unit_expr = !!get_unit_expr
  )

### PR DESCRIPTION
**Pull request by Siddhesh2097**
…versesdtm

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/advs.Rmd [hunk 1]
```diff
@@ -245,7 +245,7 @@ ratios. For instance, the function
 can be used to derive the waist-hip ratio.
 
 ```{r eval=TRUE}
-advs <- advs %>%
+advs_check <- advs %>%
```
Comment (manciniedoardo) [inline vignettes/advs.Rmd:248]: Is this intended?

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Hi @Siddhesh2097 thanks for the PR. Could you please:

- Look at the failing check for code style and fokllow the instructions there to fix
- Add a NEWS.md item for this PR
- Add pharmaversesdtm to staged_depenendencies.yaml. Note that this is why the other checks are failing: currently they install the version of pharmaversesdtm on CRAN, not the one on main, and so the new datasets are not there. See [here](https://github.com/pharmaverse/admiralonco/blob/main/staged_dependencies.yaml) for an example.

**Review by AndersAskeland — CHANGES_REQUESTED** [review]
To fix #103 we need to change the dependency of `admiraldev`. This can be done by running: usethis::use_package("admiraldev", min_version = "1.2.0")

File: vignettes/advs.Rmd (hunk 1 above)
Comment (Siddhesh2097) [reply vignettes/advs.Rmd:248]: No, updated

[the remark this answers is not in the document]
Comment (Siddhesh2097) [reply vignettes/advs.Rmd:248]: Missed to remove

File: vignettes/adcoeq.Rmd [hunk 2]
```diff
@@ -61,8 +61,8 @@ In this vignette we use example data for the CoEQ.
 The example `QS` data (`qs_metabolic`) is included in the `{admiralmetabolic}` package.
 
 ```{r, message=FALSE, warning=FALSE}
-dm_metabolic <- admiralmetabolic::dm_metabolic
-qs_metabolic <- admiralmetabolic::qs_metabolic
+dm_metabolic <- pharmaversesdtm::dm_metabolic
+qs_metabolic <- pharmvaseresdtm::qs_metabolic
```
Comment (yurovska) [inline vignettes/adcoeq.Rmd:65]:
```suggestion
qs_metabolic <- pharmaversesdtm::qs_metabolic
```

Comment (Siddhesh2097) [reply vignettes/adcoeq.Rmd:65]: Updated in last commit.

**Comment by Siddhesh2097** [comment]
LGTM now.

File: vignettes/adcoeq.Rmd [hunk 3]
```diff
@@ -61,8 +61,8 @@ In this vignette we use example data for the CoEQ.
 The example `QS` data (`qs_metabolic`) is included in the `{admiralmetabolic}` package.
```
Comment (yurovska) [inline vignettes/adcoeq.Rmd:61]: Not anymore

**Review by yurovska — APPROVED** [review]
Looks good to me. Thanks @Siddhesh2097 !

**Review by AndersAskeland — APPROVED** [review]
Great work on this @Siddhesh2097 ! 🥇 

Looks good to merge. We need to remember to update the required `pharmaversesdtm` package version when that it released - I'll make an issue for that.
